sklearn/metrics/cluster/_dbcv_internals.py

```diff
@@ -0,0 +1,316 @@
+    """
+    if metric == "precomputed":
+        if d is None:
+            raise ValueError("If metric is precomputed a d value must be provided!")
+        distance_matrix = X[labels == cluster_id, :][:, labels == cluster_id]
+    else:
+        subset_X = X[labels == cluster_id, :]
+        distance_matrix = pairwise_distances(subset_X, metric=metric, **kwd_args)
+        if d is not None:
+            msg = (
+                'The "d" value you provided is being ignored. '
+                "It's only required for precomputed distances"
+            )
+            warnings.warn(msg, UserWarning)
+        d = X.shape[1]
+
+    if no_coredist:
+        return distance_matrix, None
+
+    else:
+        core_distances = _all_points_core_distance(distance_matrix.copy(), d=d)
+        core_dist_matrix = np.tile(core_distances, (core_distances.shape[0], 1))
+        stacked_distances = np.dstack(
+            [distance_matrix, core_dist_matrix, core_dist_matrix.T]
+        )
+
+        if print_max_raw_to_coredist_ratio:
+            print(
+                "Max raw distance to coredistance ratio: "
+                + str(_max_ratio(stacked_distances))
+            )
+
+        return stacked_distances.max(axis=-1), core_distances
+
+
+def _all_points_core_distance(distance_matrix, d=2):
+    """
+    Compute the all-points-core-distance for all the points of a cluster.
+
+    Parameters
+    ----------
+    distance_matrix : array (cluster_size, cluster_size)
+        The pairwise distance matrix between points in the cluster.
+
+    d : int
+        The dimension of the data set, which is used in the computation
+        of the all-point-core-distance as per the paper.
+
+    Returns
+    -------
+    core_distances : array (cluster_size,)
+        The all-points-core-distance of each point in the cluster
+
+    References
+    ----------
+    .. [1] `Moulavi, D., Jaskowiak, P.A., Campello, R.J., Zimek, A. and Sander, J.,
+    2014. Density-Based Clustering Validation. In SDM (pp. 839-847).
+    <https://www.dbs.ifi.lmu.de/~zimek/publications/SDM2014/DBCV.pdf>`_
+    """
+    non_diagonal_indices = ~np.eye(*distance_matrix.shape, dtype=bool)
+    if distance_matrix[non_diagonal_indices].min() == 0.0:
+        raise ValueError(
+            "Identified identical points in the input data. Such points cause their "
+            "all-points-core-distances to be undefined, in turn affecting the whole "
+            "DBCV score to be undefined for the given input. (Hint: try calling "
+            "`dbcv_score` with `mst_raw_dist` set to `True`, a variant of DBCV which "
+            "does not have the aforementioned limitation due to avoiding the use of "
+            "core distances altogether.)"
+        )
+
+    distance_matrix[non_diagonal_indices] = (
+        1.0 / distance_matrix[non_diagonal_indices]
+    ) ** d
+    result = distance_matrix.sum(axis=1) / (distance_matrix.shape[0] - 1)
+    return result**-1.0 / d
```

Comment:
I think that this is currently equivalent to `1 / (result * d)`
```suggestion
    return result ** (-1 / d)
```

Reply:
You're correct, I apologize for the embarrassing oversight on my part, glad you caught this now!

That explains the weird formatting the linter was enforcing here.

The precedence difference between simple assignment and the previous `/=` means parens are required, as you suggest, I'll apply it straight away.